### PR TITLE
Color management with CSS Variables & HSL

### PR DESCRIPTION
**Pull request by sashachabin**
## TL;DR

1. Now we works with colors in `variables.css`. 

2. We use dynamic colors in HSL:
`hsl( var(--color-h) var(--color-s) calc(var(--color-l) + 20%) / 10%)`

3. I removed a huge amount of `style=""` and update some components

Colors

## Updates
- [x] Switch dynamic colors froms JS functions to HSL
- [x] Move all color variables to CSS Variables in `variables.css`
- [x] Move Notion text color-tokens to `variables.css`

## Components changes
- [x] Remove all inline styles with colors
- [x] Update link colors in pages
- [x] Change colors in `<SidePage>` (search control)

## Removed dependencies
- [x] Remove npm package `color`
- [x] Remove npm package `rgba-to-rgb`

## Additionally refactoring
- [x] Separate `variables.css` from `globals.css`
- [x] Rename `<CommonLinks>` → `<Ecosystem>`
- [x] Remove inline `borderRadius` huge customization fragment in `tableOfContentsLink`

### CODE REVIEW

File: components/TableOfContents/TableOfContents.jsx [hunk 1]
```diff
@@ -84,21 +65,6 @@ function TableOfContents({ tableOfContentArr, currentPageUrl = [], anchorLinks,
                         pathname: '/[[...pageUrl]]',
                         query: { pageUrl: [currentPageUrl[0], url] },
                     }}
-                    style={{
```
Comment (sashachabin) [inline components/TableOfContents/TableOfContents.jsx:87]: Movee borderRadius customization to CSS

**Comment by sashachabin** [comment]
@Omelekhin 

Now we can change colors in browser

image

**Comment by sashachabin** [comment]
@Omelekhin 

Example for common colors

image

**Comment by sashachabin** [comment]
- [ ] Fix colors of search results

**Review by fidlow — COMMENTED** [review]
Approve!
